connectivity/tests/perfpod.go

```diff
@@ -0,0 +1,120 @@
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium-cli/defaults"
+)
+
+// TCP Network Performance
+func TCPPodtoPod(n string) check.Scenario {
+	return &tcpPodToPod{
+		name: n,
+	}
+}
+
+type tcpPodToPod struct {
+	name string
+}
+
+func (s *tcpPodToPod) Name() string {
+	tn := "perf[tcp]-pod-to-pod"
+	if s.name == "" {
+		return tn
+	}
+	return fmt.Sprintf("%s:%s", tn, s.name)
+}
+
+func (s *tcpPodToPod) Run(ctx context.Context, t *check.Test) {
+	for _, c := range t.Context().PerfClientPods() {
+		c := c
+		for _, server := range t.Context().PerfServerPod() {
+			action := t.NewAction(s, "iperf-tcp", &c, server)
+			action.CollectFlows = false
+			action.Run(func(a *check.Action) {
+				iperfStream(ctx, server.Pod.Status.PodIP, c.Pod.Name, action, t.Context().PerfResults, 1, false)
+			})
+		}
+	}
+}
+
+// UDP Network pPerformance
+func UDPPodtoPod(n string) check.Scenario {
+	return &udpPodtoPod{
+		name: n,
+	}
+}
+
+type udpPodtoPod struct {
+	name string
+}
+
+func (s *udpPodtoPod) Name() string {
+	tn := "perf[udp]-pod-to-pod"
+	if s.name == "" {
+		return tn
+	}
+	return fmt.Sprintf("%s:%s", tn, s.name)
+}
+
+func (s *udpPodtoPod) Run(ctx context.Context, t *check.Test) {
+	for _, c := range t.Context().PerfClientPods() {
+		c := c
+		for _, server := range t.Context().PerfServerPod() {
+			action := t.NewAction(s, "iperf-udp", &c, server)
+			action.CollectFlows = false
+			action.Run(func(a *check.Action) {
+				a.CollectFlows = false
+				iperfStream(ctx, server.Pod.Status.PodIP, c.Pod.Name, a, t.Context().PerfResults, 1, true)
+			})
+		}
+	}
+}
+
+func iperfStream(ctx context.Context, sip string, podname string, a *check.Action, result map[string]map[string]string, samples int, udp bool) {
+	// Allow the user to override aspects of the test
+	iteration, _ := strconv.Atoi(os.Getenv("samples"))
+	duration, _ := strconv.Atoi(os.Getenv("duration"))
+
+	if samples < iteration {
+		samples = iteration
+	}
+	if duration <= 0 {
+		duration = defaults.PerfTestDuration
+	}
+	for i := 0; i < samples; i++ {
+		// Result data
+		r := make(map[string]string)
+
+		exec := []string{"/usr/bin/iperf3", "-c", sip, "-t", fmt.Sprintf("%d", duration), "-J"}
+		if udp {
+			exec = []string{"/usr/bin/iperf3", "-u", "-c", sip, "-t", fmt.Sprintf("%d", duration), "-J"}
+		}
```

Comment:
This can be moved out of the `for` loop, I think. It only depends on arguments passed to the function.